lib/arel/enhance/context_enhancer/arel_table.rb

```diff
@@ -62,6 +64,9 @@ def self.call(node)
           elsif parent_object.is_a?(Arel::Nodes::Into)
             context[:alias] = true
 
+          elsif parent_object.is_a?(Arel::Nodes::TableAlias)
+            context[:range_variable] = true
+
```

Comment:
I'm not 100% confident this is specific enough 🤔, could you add another check for the `node.parent.parent`?